hudi-common/src/main/java/org/apache/hudi/common/util/collection/DiskBasedMap.java

```diff
@@ -395,4 +417,48 @@ public int compareTo(ValueMetadata o) {
       return Long.compare(this.offsetOfValue, o.offsetOfValue);
     }
   }
+
+  private static class DiskCompressionInstance implements Serializable {
+    public static final int DISK_COMPRESSION_INITIAL_BUFFER_SIZE = 1048576;
+
+    // Caching ByteArrayOutputStreams to avoid recreating it for every operation
+    private final ByteArrayOutputStream compressBaos;
+    private final ByteArrayOutputStream decompressBaos;
+    private final byte[] decompressBuffer;
+
+    DiskCompressionInstance() {
+      compressBaos = new ByteArrayOutputStream(DISK_COMPRESSION_INITIAL_BUFFER_SIZE);
+      decompressBaos = new ByteArrayOutputStream(DISK_COMPRESSION_INITIAL_BUFFER_SIZE);
+      decompressBuffer = new byte[8192];
```

Comment:
pull this into a static final?

Reply:
what if there is a value beyond 8192? better to use something that is dynamic?

Reply:
Yes, it is dynamic and uses ByteArrayOutputStream (decompressBaos). decompressBuffer is just a intermediate buffer that reads from the inputStream and writes to the outputstream, aka decompressBaos. So 8192 is the max bytes read at a given time.